### ISSUE TITLE
new_tunnel should support NULL preshared_key

### ISSUE THREAD

**Issue by saurik** [cloudflare/boringtun#210]
The API for new_tunnel was recently changed in a133f1df70e09052f3fabdcd50cba98d299b0e51 to support setting the preshared key from C. As None is a valid preshared key, one would expect NULL to set that, but instead the process crashes dereferencing the null pointer. You can pass an invalid string, which gets converted to None, which works for now but that's also weird as one would expect an invalid struct should return a null tunnel instead of just silently changing the key to None.

**Comment by wiresock** [comment]
I tend to agree. However, meanwhile it should not be an invalid string, it can be an empty string ("") instead of NULL.

**Comment by saurik** [comment]
Right. I'm saying that, right now, passing any invalid string--including ""--is being converted to None and silently ignored--which I think is strange and incorrect behavior--while NULL is crashing the process (which also seems wrong). I'm saying that the behavior which would make sense is for NULL to map to None and all invalid strings to cause the function to fail and return NULL (instead of the new tunnel): if you pass a string it should be required to be valid for the function to work correctly.

**Comment by Noah-Kennedy** [comment]
I've opened up a PR to fix this. @saurik @wiresock could you take a look at it?

**Comment by wiresock** [comment]
Thanks! I have tested on Windows and I can confirm that it works as expected.
